core/src/main/scala/chisel3/Aggregate.scala

```diff
@@ -184,7 +184,10 @@ sealed class Vec[T <: Data] private[chisel3] (gen: => T, val length: Int) extend
   }
 
   private[chisel3] override def bind(target: Binding, parentDirection: SpecifiedDirection): Unit = {
-    _parent.foreach(_.addId(this))
+    // ConstrainedBinding means this is scoped to a Module, no need to add to _ids otherwise
+    if (target.isInstanceOf[ConstrainedBinding]) {
+      _parent.foreach(_.addId(this))
+    }
```

Comment:
This snippet is repeated multiple times. Is there a factoring that avoids the repetition?

Reply:
Good point, I'll factor it out.